routing/http/server/server.go

```diff
@@ -78,8 +86,16 @@ type ContentRouter interface {
 	// PutIPNS stores the provided [ipns.Record] for the given [ipns.Name].
 	// It is guaranteed that the record matches the provided name.
 	PutIPNS(ctx context.Context, name ipns.Name, record *ipns.Record) error
+
+	// GetClosestPeers returns the DHT closest peers to the given peer ID.
+	// If empty, it will use the content router's peer ID (self). `closerThan` (optional) forces resulting records to be closer to `PeerID` than to `closerThan`. `count` specifies how many records to return ([1,100], with 20 as default when set to 0).
+	GetClosestPeers(ctx context.Context, peerID peer.ID) (iter.ResultIter[*types.PeerRecord], error)
```

Comment:
```suggestion
	GetClosestPeers(ctx context.Context, key string) (iter.ResultIter[*types.PeerRecord], error)
```

As per https://github.com/ipfs/specs/pull/476#discussion_r2383524824, we can use the string representation of the peer id or multihash as in [go-libp2p-kad-dht](https://github.com/libp2p/go-libp2p-kad-dht/blob/6e9a59c05736c7b40586ff385c5efc96981b46f8/lookup.go#L23) ([peer id](https://github.com/libp2p/go-libp2p-kad-dht/blob/6e9a59c05736c7b40586ff385c5efc96981b46f8/routing.go#L642), [CID](https://github.com/libp2p/go-libp2p-kad-dht/blob/6e9a59c05736c7b40586ff385c5efc96981b46f8/routing.go#L566), [IPNS key](https://github.com/libp2p/go-libp2p-kad-dht/blob/6e9a59c05736c7b40586ff385c5efc96981b46f8/routing.go#L107))

Reply:
Note: string representation here is "casting" as string, not doing .String().

Reply:
Triage note:
- `string` is too risky ambigous -- will elad to hard to debug bugs
- use `CID` type and extract raw bytes via vasting bytes→string internally

Reply:
We now use

```go
GetClosestPeers(ctx context.Context, key cid.Cid) (iter.ResultIter[*types.PeerRecord], error)
```

(the Cid acts as a container for the Multihash digest, which then is casted as string to kad-dht for legacy reasons).